lib/core/versions/0.9.0/util/Compressor.ts

```diff
@@ -23,13 +27,44 @@ export default class Compressor {
 
   /**
    * Decompresses the input and returns it as buffer.
-   * @param inputAsBuffer The gzip compressed data.
    */
-  public static async decompress (inputAsBuffer: Buffer): Promise<Buffer> {
+  public static async decompress (inputAsBuffer: Buffer, maxAllowedDecompressedSizeInBytes: number): Promise<Buffer> {
+    // Create a gunzip transform object.
+    const gunzip = zlib.createGunzip();
 
-    const result = await Compressor.gUnzipAsync(inputAsBuffer);
+    let content = Buffer.alloc(0);
 
-    // Casting result to Buffer as that's what is returned by gzip
-    return result as Buffer;
+    // Handle the data chunks that are decompressed as they come in.
+    gunzip.on('data', (chunk: Buffer) => {
```

Comment:
Do we have to worry about chunk.length being huge? Or is there a default size for chunk that gunzip controls

Reply:
There is a default chunk size 16KB, and I added a test for this exact concern.